tests/test_command_pipeline.py

```diff
@@ -0,0 +1,161 @@
+@patch("boto3.client")
+def test_pipeline_generate_overwrites_any_existing_config_files(
+    mocked_boto3_client, tmp_path, switch_to_tmp_dir_and_copy_fixtures
+):
+    mock_codestar_connections_boto_client(mocked_boto3_client, ["test-app"])
+    setup_git_repository()
+    buildspec, cfn_patch, manifest = setup_output_file_paths(tmp_path)
+    for path in [buildspec, cfn_patch, manifest]:
+        os.makedirs(path.parent, exist_ok=True)
+        with open(path, "w") as fh:
+            print("Pre-existing file contents", file=fh)
+
+    CliRunner().invoke(generate_config)
+
+    assert_output_file_contents_match_expected(buildspec, "buildspec.yml")
+    assert_output_file_contents_match_expected(manifest, "manifest.yml")
+    assert_output_file_contents_match_expected(cfn_patch, "overrides/cfn.patches.yml")
+
+
+@patch("boto3.client")
+def test_pipeline_generate_with_no_codestar_connection_exits_with_message(
+    mocked_boto3_client, switch_to_tmp_dir_and_copy_fixtures
+):
+    mock_codestar_connections_boto_client(mocked_boto3_client, [])
+    setup_git_repository()
+
+    result = CliRunner().invoke(generate_config)
+
+    assert result.exit_code == 1
+    assert "Error: There is no CodeStar Connection to use" in result.output
+
+
+def test_pipeline_generate_with_no_repo_fails_with_message(switch_to_tmp_dir_and_copy_fixtures):
+    result = CliRunner().invoke(generate_config)
+
+    assert result.exit_code == 1
+    assert "Error: The current directory is not a git repository" in result.output
+
+
+def test_pipeline_generate_with_no_pipeline_yml_fails_with_message(
+    switch_to_tmp_dir_and_copy_fixtures,
+):
+    os.remove("pipelines.yml")
+
+    result = CliRunner().invoke(generate_config)
+
+    assert result.exit_code == 1
+    assert "Error: There is no pipelines.yml" in result.output
+
+
+def test_pipeline_generate_pipeline_yml_invalid_fails_with_message(
+    switch_to_tmp_dir_and_copy_fixtures,
+):
+    with open("pipelines.yml", "w") as fh:
+        print("{invalid data", file=fh)
+
+    result = CliRunner().invoke(generate_config)
+
+    assert result.exit_code == 1
+    assert "Error: The pipelines.yml file is invalid" in result.output
+
+
+def test_pipeline_generate_with_no_bootstrap_yml_fails_with_message(
+    switch_to_tmp_dir_and_copy_fixtures,
+):
+    os.remove("bootstrap.yml")
+
+    result = CliRunner().invoke(generate_config)
+
+    assert result.exit_code == 1
+    assert "Error: There is no bootstrap.yml" in result.output
+
+
+def test_pipeline_generate_bootstrap_yml_invalid_fails_with_message(
+    switch_to_tmp_dir_and_copy_fixtures,
+):
+    with open("bootstrap.yml", "w") as fh:
+        print("{invalid data", file=fh)
+
+    result = CliRunner().invoke(generate_config)
+
+    assert result.exit_code == 1
+    assert "Error: The bootstrap.yml file is invalid" in result.output
+
+
+def assert_file_created_in_stdout(output_file, result, tmp_path):
```

Comment:
Think this and below should live in a test utils file somewhere

Reply:
I have moved them to `conftest.py`, though that probably wants breaking up, but not in this pull request which has enough noise already.